src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/StorageBlobScanInfoManager.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Storage.Blob;
+using Microsoft.WindowsAzure.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Host.Blobs.Listeners
+{
+    internal class StorageBlobScanInfoManager : IBlobScanInfoManager
+    {
+        private readonly JsonSerializer _serializer;
+        private readonly string _hostId;
+        private IStorageBlobDirectory _blobScanInfoDirectory;
+
+        public StorageBlobScanInfoManager(string hostId, IStorageBlobClient blobClient)
+        {
+            if (string.IsNullOrEmpty(hostId))
+            {
+                throw new ArgumentNullException("hostId");
+            }
+            if (blobClient == null)
+            {
+                throw new ArgumentNullException("blobClient");
+            }
+
+            _hostId = hostId;
+            JsonSerializerSettings settings = new JsonSerializerSettings
+            {
+                DateFormatHandling = DateFormatHandling.IsoDateFormat
+            };
+            _serializer = JsonSerializer.Create(settings);
+
+            string blobScanInfoDirectoryPath = string.Format(CultureInfo.InvariantCulture, "blobscaninfo/{0}", _hostId);
+            _blobScanInfoDirectory = blobClient.GetContainerReference(HostContainerNames.Hosts).GetDirectoryReference(blobScanInfoDirectoryPath);
+        }
+
+        public async Task<DateTime?> LoadLatestScanAsync(string storageAccountName, string containerName)
+        {
+            IStorageBlockBlob scanInfoBlob = GetScanInfoBlobReference(storageAccountName, containerName);
+            DateTime? latestScan = null;
+            try
+            {
+                string scanInfoLine = await scanInfoBlob.DownloadTextAsync(CancellationToken.None);
+                ScanInfo scanInfo;
+                using (StringReader stringReader = new StringReader(scanInfoLine))
+                {
+                    scanInfo = (ScanInfo)_serializer.Deserialize(stringReader, typeof(ScanInfo));
+                }
+
+                if (scanInfo != null)
+                {
+                    latestScan = scanInfo.LatestScan;
```

Comment:
nit: do we have access to the `?.` accessor?  Could shorten a fair bit of null checking